yt/yt/server/query_tracker/handler_base.cpp

```diff
@@ -376,6 +384,31 @@ bool TQueryHandlerBase::TryWriteQueryState(EQueryState state, EQueryState previo
         }
         WaitFor(transaction->Commit())
             .ThrowOnError();
+
+        auto now = TInstant::Now();
+        StateTimes[previousState] += now - LastStateChange_.value_or(record.StartTime);
+        LastStateChange_ = now;
+
+        if (state == EQueryState::Completing || state == EQueryState::Failing) {
```

Comment:
why don't we emit sensor values as soon as they are obtained, and instead firts collect all the  StateTimes in the separate variable until query completion?

Reply:
For most counters, the situation is as you describe. However, for example, in the case of a pending state, a query may enter into this state multiple times (if yqla is overloaded)